### ISSUE TITLE
`sympy.simplify` expands out `log`s.

### ISSUE THREAD

**Issue by DUOLabs333** [sympy/sympy#23210]
For example, `simplify(4000*log(5))` gives me `log(<very big number>)`.

**Comment by oscargus** [comment]
I can see the issue, but the "problem" is that `log(z)` is simpler than `x*log(y)` when `z = y**x` can be written as a single number.

```
simplify(4000*log(5)).count_ops()
(4000*log(5)).count_ops()
```

`simplify` does have a keyword argument, `measure` where you can pass a function to determine which expression is simpler. So if you can figure out how to measure which one of these are simpler, you can use that.

Maybe one should add some logic to determine if really x**y should evaluate numerically, but I guess that will cause other issues, like if you dividing with a large number. And as ints are exact in Python (and SymPy), there will not be any real problems evaluating it, except then for printing it.

(The whole idea of `simplify` is basically applying different rewrites and see if the resulting expression is simpler, in some heuristic order.)

**Comment by oscarbenjamin** [comment]
> And as ints are exact in Python (and SymPy), there will not be any real problems evaluating it,

There definitely can be problems evaluating it e.g.:
```python
In [74]: for n in range(1, 10):
    ...:     simplify((10**n)*log(10**n))
    ...:     print(n)
    ...: 
1
2
3
4
5
6
^C---------------------------------------------------------------------------
KeyboardInterrupt
```
See #6835 and the many linked issues.

**Comment by oscargus** [comment]
> There definitely can be problems evaluating it e.g.:

Let me rephrase that: there are rarely any numerical problems evaluating it.

**Comment by oscargus** [comment]
Of course, if you solve a problem with gazillions of digits, it will naturally be slow.

**Comment by DUOLabs333** [comment]
@oscargus Does `sympy` have a default `measure` function?

**Comment by DUOLabs333** [comment]
I'm trying to modify `count_ops` to fix this, but when trying to use it, even though I defined `count_ops`, I get `NameError: name 'count_ops' is not defined`

**Comment by oscarbenjamin** [comment]
Personally I still think that the `simplify` behaviour should be changed here. Really large numbers should be avoided where possible because they can slow things down a lot.
